ggml/src/ggml-vulkan/ggml-vulkan.cpp

```diff
@@ -11189,6 +11186,8 @@ static bool ggml_backend_vk_device_supports_op(ggml_backend_dev_t dev, const ggm
         case GGML_OP_OPT_STEP_ADAMW:
         case GGML_OP_OPT_STEP_SGD:
             return op->src[0]->type == GGML_TYPE_F32;
+        case GGML_OP_ARGSORT:
+            return op->ne[0] <= max_argsort_cols;
```

Comment:
Wouldn't it be better to fall back to the slower shader instead of disabling support if rows become too big?

Reply:
Sizes larger than 1024 never worked. The shared memory array isn't large enough to hold the indices, and also the assert `GGML_ASSERT(ncols_pad <= 1024)` fires. Also would not work in the CUDA backend, I think. In my testing I think I only saw row length of 32 or 64 in real models.